apps/taxes/src/modules/taxes/providers/avatax/ui/avatax-configuration-form.tsx

```diff
@@ -0,0 +1,312 @@
+
+  const { mutate: createMutation, isLoading: isCreateLoading } =
+    trpcClient.providersConfiguration.create.useMutation({
+      onSuccess({ id }) {
+        setInstanceId(id);
+        refetchProvidersConfigurationData();
+        appBridge?.dispatch(
+          actions.Notification({
+            title: "Success",
+            text: "Saved app configuration",
+            status: "success",
+          })
+        );
+      },
+      onError(error) {
+        appBridge?.dispatch(
+          actions.Notification({
+            title: "Error",
+            text: error.message,
+            status: "error",
+          })
+        );
+      },
+    });
+
+  const { mutate: updateMutation, isLoading: isUpdateLoading } =
+    trpcClient.providersConfiguration.update.useMutation({
+      onSuccess() {
+        refetchProvidersConfigurationData();
+        appBridge?.dispatch(
+          actions.Notification({
+            title: "Success",
+            text: "Updated Avalara configuration",
+            status: "success",
+          })
+        );
+      },
+      onError(error) {
+        appBridge?.dispatch(
+          actions.Notification({
+            title: "Error",
+            text: error.message,
+            status: "error",
+          })
+        );
+      },
+    });
+
+  const { mutate: deleteMutation } = trpcClient.providersConfiguration.delete.useMutation({
+    onSuccess() {
+      resetInstanceId();
+      refetchProvidersConfigurationData();
+      refetchChannelConfigurationData();
+      appBridge?.dispatch(
+        actions.Notification({
+          title: "Success",
+          text: "Remove TaxJar instance",
+          status: "success",
+        })
+      );
+    },
+    onError(error) {
+      appBridge?.dispatch(
+        actions.Notification({
+          title: "Error",
+          text: error.message,
+          status: "error",
+        })
+      );
+    },
+  });
+
+  const textFieldProps: TextFieldProps = {
+    fullWidth: true,
+  };
+
+  const onSubmit = (values: FormValues) => {
+    if (instanceId) {
+      updateMutation({
+        id: instanceId,
+        provider: {
+          ...values,
+          provider: "avatax",
+        },
+      });
+    } else {
+      createMutation({
+        provider: {
+          ...values,
+          provider: "avatax",
+        },
+      });
+    }
+  };
+
+  const closeWarningDialog = () => {
+    setIsWarningDialogOpen(false);
+  };
+
+  const openWarningDialog = () => {
+    setIsWarningDialogOpen(true);
+  };
+
+  const deleteProvider = () => {
+    closeWarningDialog();
+    if (instanceId) {
+      deleteMutation({ id: instanceId });
+    }
+  };
+
+  const isLoading = isCreateLoading || isUpdateLoading;
+
+  return (
+    <>
+      <form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <Controller
+              name="name"
+              control={control}
+              defaultValue=""
```

Comment:
Put initial values in form declaration and reuse the same values as container empty defaults: 
```
const { register, handleSubmit, control, setValue, getValues, reset } =
    useForm<MjmlConfiguration>({
      defaultValues: props.initialValues || getDefaultEmptyConfiguration(),
    });
```